test/vfd_swmr_indep_rw_writer.c

```diff
@@ -0,0 +1,788 @@
+
+        if (dset_id >= 0)
+            break;
+        else
+            decisleep(s->check_interval);
+    }
+
+    s->r_dsetid = dset_id;
+    if (i == NUM_ATTEMPTS) {
+        HDfprintf(stderr, "dataset opening reachs the maximal number of attempts\n");
+        TEST_ERROR;
+    }
+
+    if (s->first_proc)
+        dbgf(1, "Process 0: Successfully open the dataset %s for file %s.\n", dname, s->filename[fopen_idx]);
+    else
+        dbgf(1, "Process 1: Successfully open the dataset %s for file %s.\n", dname, s->filename[fopen_idx]);
+    return true;
+
+error:
+    return false;
+}
+
+static bool
+vrfy_dset(const state_t *s, mat_t *mat)
+{
+
+    unsigned int i;
+    unsigned int which = 0;
+    herr_t       status;
+
+    /* A failure to read the data may indicate the data isn't ready yet.  Instead of displaying the error
+     * stack, simply return false and let the caller repeat this step.
+     */
+
+    for (i = 0; i < NUM_ATTEMPTS; i++) {
+        H5E_BEGIN_TRY
+        {
+            status = H5Dread(s->r_dsetid, H5T_NATIVE_UINT32, H5S_ALL, H5S_ALL, H5P_DEFAULT, mat->elt);
+        }
+        H5E_END_TRY;
+        if (status < 0) {
+            decisleep(1);
+            /* Refresh the dataset and try it again */
+            if (H5Drefresh(s->r_dsetid) < 0) {
+                HDfprintf(stderr, "H5Drefresh failed\n");
+                TEST_ERROR;
+            }
+        }
+        else
+            break;
+    }
+    if (i == NUM_ATTEMPTS) {
+        HDfprintf(stderr, "dataset verification reached the maximal number of attempts\n");
+        TEST_ERROR;
+    }
+    if (s->first_proc)
+        which = 1;
+
+    if (verify_matrix(mat, which) == false) {
+        HDfprintf(stderr, "dataset verification failed\n");
+        TEST_ERROR;
+    }
+
+    if (s->first_proc)
+        dbgf(1, "Process 0: Successfully verify a dataset.\n");
+    else
+        dbgf(1, "Process 1: Successfully verify a dataset.\n");
+
+    return true;
+
+error:
+    return false;
+}
+
+static bool
+read_vrfy_dataset(state_t *s, mat_t *mat)
+{
+
+    if (false == open_dset(s)) {
+        HDfprintf(stderr, "Reader: open_dataset() failed\n");
+        TEST_ERROR;
+    }
+    if (false == vrfy_dset(s, mat)) {
+        HDfprintf(stderr, "Reader: vrfy_dataset() failed\n");
+        TEST_ERROR;
+    }
+
+    if (H5Dclose(s->r_dsetid) < 0) {
+        HDfprintf(stderr, "Reader: H5Dclose() failed\n");
+        TEST_ERROR;
+    }
+    return true;
+
+error:
+
+    H5E_BEGIN_TRY
+    {
+        H5Dclose(s->r_dsetid);
+    }
+    H5E_END_TRY;
+    return false;
+}
+
+static bool
+close_pl(const state_t *s)
+{
+
+    if (H5Pclose(s->fapl) < 0) {
+        printf("H5Pclose failed\n");
+        TEST_ERROR;
+    }
+
+    if (H5Pclose(s->fcpl) < 0) {
+        printf("H5Pclose failed\n");
+        TEST_ERROR;
+    }
+
+    return true;
+
+error:
+    H5E_BEGIN_TRY
+    {
+        H5Pclose(s->fapl);
+        H5Pclose(s->fcpl);
+    }
+    H5E_END_TRY;
+
+    return false;
+}
+
```

Comment:
Is this test program going to be modified further to accommodate remaining items for #A, #B and #C in the task list?

Reply:
I am not sure if this testing program will be modified. This one is the first one.  For the task list,  John told me the information on the list  is not exactly what's in his mind. So if we do need to modify, it may not be according to the task list.